src/main.rs

```diff
@@ -241,6 +241,14 @@ struct Args {
     /// Overrides the OpenTelemetry endpoint used for emitting logs. This can also be set with `OTEL_EXPORTER_OTLP_LOGS_ENDPOINT`.
     #[clap(long = "override-logs-endpoint", hide = true)]
     logs_endpoint: Option<String>,
+
+    /// Configures whether grpc or http will be used for exporting the enabled telemetry. This defaults to 'http'.
+    #[clap(
+        long = "observability-protocol",
+        env = "WASMCLOUD_OBSERVABILITY_PROTOCOL",
+        hide = true
+    )]
+    observability_protocol: Option<String>,
```

Comment:
Here I think this should definitely be an `Option<OtelProtocol>`, clap should be able to handle the parsing for you well only accepting `grpc` or `http` and failing at the command line instead of later on

Reply:
Yeah, you can probably move your `parse` function to be an implementation of `FromStr` for OtelProtocol and I'm pretty sure clap will just use that

Reply:
This is exactly what I wanted, thanks for the pointer, I had implemented `From<Option<String>>` which didn't give me the option to return an Error in the case that the supplied value couldn't be parsed to something.